### PR TITLE
GEODE-8643: Fix IllegalArgumentException in ParallelGatewaySenderQueu…

### PR DESCRIPTION
**Pull request by albertogpz**
…e for negative sleep value

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

**Review by kohlmu-pivotal — CHANGES_REQUESTED** [review]

File: geode-core/src/main/java/org/apache/geode/internal/cache/wan/parallel/ParallelGatewaySenderQueue.java [hunk 1]
```diff
@@ -1393,7 +1393,11 @@ private boolean areAllTransactionsCompleteInBatch(Map incompleteTransactions) {
     return (incompleteTransactions.size() == 0);
   }
 
-  private long getTimeToSleep(long timeToWait) {
+  @VisibleForTesting
+  public static long getTimeToSleep(long timeToWait) {
```
[the remark this answers is not in the document]
Comment (albertogpz) [reply geode-core/src/main/java/org/apache/geode/internal/cache/wan/parallel/ParallelGatewaySenderQueue.java:1397]: Good points, thanks.

File: geode-core/src/main/java/org/apache/geode/internal/cache/wan/parallel/ParallelGatewaySenderQueue.java [hunk 2]
```diff
@@ -1393,7 +1393,11 @@ private boolean areAllTransactionsCompleteInBatch(Map incompleteTransactions) {
...
+    if (timeToWait < 0) {
```
[the remark this answers is not in the document]
Comment (albertogpz) [reply geode-core/src/main/java/org/apache/geode/internal/cache/wan/parallel/ParallelGatewaySenderQueue.java:1398]: Agree.

**Comment by kohlmu-pivotal** [comment]
@albertogpz please squash into single commit. Other than that LGTM

**Comment by albertogpz** [comment]
> @albertogpz please squash into single commit. Other than that LGTM

Thanks, @kohlmu-pivotal. I prefer that the squash is done at the final Squash and Merge to develop as it has been suggested in the community in e-mails. I have always done it that way with the exception of commits sent to fix tests.

**Review by kohlmu-pivotal — APPROVED** [review]

**Review by mkevo — APPROVED** [review]

**Pull request merged**